backend/main/src/Group/Query/FindContainersOfAUserQuery/FindContainersOfAUserQuery.ts

```diff
@@ -0,0 +1,113 @@
+import { IQuery } from "src/Shared/Layers/Query/IQuery";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import {
+	ContainerData,
+	UsersGroup,
+} from "src/Shared/Adapters/DB/NishikiDBTypes";
+import { Ok, Err, Result } from "result-ts-type";
+import { QueryError } from "src/Shared/Utils/Errors";
+import { isValidUUIDV4 } from "src/Shared/Utils/Validator";
+
+export class FindContainersOfAUserQuery
+	implements IQuery<{ userId: string }, IContainerData, InvalidUUId>
+{
+	constructor(private readonly nishikiDynamoDBClient: NishikiDynamoDBClient) {}
+
+	public async execute(input: { userId: string }): Promise<
+		Result<IContainerData, InvalidUUId>
+	> {
+		const { userId } = input;
+		if (!isValidUUIDV4(userId)) {
+			return Err(new InvalidUUId("User ID is invalid"));
+		}
+
+		const userGroups =
+			await this.nishikiDynamoDBClient.listOfUsersGroup(userId);
+
+		if (userGroups.length === 0) {
+			return Ok({ containers: [] });
+		}
+
+		const groupContainerDetailsList = (
+			await Promise.all(
+				userGroups.map(async (userGroup) => {
+					return await this.getGroupContainerDetails(userGroup);
+				}),
+			)
+		).flat();
+
+		return Ok({ containers: groupContainerDetailsList });
+	}
+
+	private async getGroupContainerDetails(
+		userGroup: UsersGroup,
+	): Promise<IContainer[]> {
+		const [groupData, containerIds] = await Promise.all([
+			this.nishikiDynamoDBClient.getGroup({
+				groupId: userGroup.groupId,
+			}),
+			this.nishikiDynamoDBClient.listOfContainers(userGroup.groupId),
+		]);
+
+		if (!groupData) {
+			return [];
+		}
+		if (containerIds.length === 0) {
+			return [];
+		}
```

Comment:
You can write like this.
```suggestion
	if (!(groupData && containerIds.length > 0)) {
			return [];
		}
```

Reply:
Thanks! It looks better. I've fixed it.